src/OpenTelemetry/Trace/Builder/TracerProviderBuilderSdk.cs

```diff
@@ -33,6 +33,8 @@ public TracerProviderBuilderSdk(IServiceProvider serviceProvider)
 
     public List<string> Sources { get; } = new();
 
+    public Func<ActivitySource, bool>? SourceFilter { get; private set; } = null;
```

Comment:
The API is called "AddSource" so to fit in better with our other "Add" methods I think we should switch this to a list and allow more than one to be registered. Also I would avoid using "filter" we use that other places to mean something different. How about...

```csharp
public List<Predicate<ActivitySource>> SourceSelectionPredicates { get; } = new();
```